ballerina/tests/client_test.bal

```diff
@@ -0,0 +1,808 @@
+
+    Client graphqlClient = check new (url);
+    json|ClientError payload = graphqlClient->executeWithType(document);
+    if payload is ClientError {
+        if payload is ServerError {
+            json actualPayload = payload.detail().toJson();
+            json expectedPayload = {
+                data: {
+                    specialHolidays: ["TUESDAY", null, "THURSDAY"]
+                },
+                errors: [
+                    {
+                        message: "Holiday!",
+                        locations: [
+                            {
+                                line: 1,
+                                column: 9
+                            }
+                        ],
+                        path: ["specialHolidays", 1]
+                    }
+                ],
+                extensions: null
+            };
+            assertJsonValuesWithOrder(actualPayload, expectedPayload);
+        }
+    }
+}
+
+@test:Config {
+    groups: ["client"]
+}
+isolated function testClientExecuteWithPartialDataRequest() returns error? {
+    string url = "http://localhost:9095/special_types";
+    string document = string `query { specialHolidays }`;
+
+    Client graphqlClient = check new (url);
+    json actualPayload = check graphqlClient->execute(document);
+    json expectedPayload = {
+        errors: [
+            {
+                message: "Holiday!",
+                locations: [
+                    {
+                        line: 1,
+                        column: 9
+                    }
+                ],
+                path: ["specialHolidays", 1]
+            }
+        ],
+        data: {
+            specialHolidays: ["TUESDAY", null, "THURSDAY"]
+        }
+    };
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+@test:Config {
+    groups: ["client"]
+}
+isolated function testClientExecuteWithTypeWithMultipleOperationsWithoutOperationNameInRequest() returns error? {
+    string url = "http://localhost:9091/validation";
+    string document = check getGraphQLDocumentFromFile("multiple_operations_without_operation_name_in_request.graphql");
+
+    Client graphqlClient = check new (url);
+    json|ClientError payload = graphqlClient->executeWithType(document);
+    if payload is ClientError {
+        if payload is RequestError {
+            json actualPayload = payload.detail()?.body.toJson();
+            json expectedPayload = {
+                errors: [
+                    {
+                        message: "Must provide operation name if query contains multiple operations.",
+                        locations: []
+                    }
+                ]
+            };
+            assertJsonValuesWithOrder(actualPayload, expectedPayload);
+        } else {
+            test:assertFail("Invalid error type");
+        }
+    } else {
+        test:assertFail("Json response received for a request with multiple operations without operation name");
+    }
```

Comment:
We don't necessarily need type guard each and every type here. This might be a better way. 
```suggestion
    test:assertTrue(payload is RequestError);
    RequestError err = <RequestError>payload;
    json actualPayload = err.detail()?.body.toJson();
    json expectedPayload = {
        errors: [
            {
                message: "Must provide operation name if query contains multiple operations.",
                locations: []
            }
        ]
    };
```